src/main/kotlin/org/koitharu/kotatsu/parsers/site/uk/HentaiUkr.kt

```diff
@@ -0,0 +1,146 @@
+package org.koitharu.kotatsu.parsers.site.uk
+
+import androidx.collection.ArraySet
+import okhttp3.Interceptor
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONObject
+import org.koitharu.kotatsu.parsers.MangaLoaderContext
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.PagedMangaParser
+import org.koitharu.kotatsu.parsers.config.ConfigKey
+import org.koitharu.kotatsu.parsers.model.*
+import org.koitharu.kotatsu.parsers.util.*
+import org.koitharu.kotatsu.parsers.util.json.getStringOrNull
+import org.koitharu.kotatsu.parsers.util.json.toJSONList
+import java.text.SimpleDateFormat
+import java.util.*
+
+private const val HEADER_ENCODING = "Content-Encoding"
+private val date = SimpleDateFormat("yyyy-MM-dd", Locale.US)
+@MangaSourceParser("HENTAIUKR", "Hentaiukr", "uk")
+class HentaiUkrParser(context: MangaLoaderContext) : PagedMangaParser(context, MangaSource.HENTAIUKR, 1),
+	Interceptor {
+
+	private val allManga get() = "https://$domain/search/objects.json"
+
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("hentaiukr.com")
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.NEWEST,
+	)
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val json= webClient.httpGet(allManga).parseJson().getJSONArray("manga").toJSONList().find { it.getString("name") == manga.title }
+		val html_manga = webClient.httpGet("https://$domain${manga.url}").parseHtml()
+		val about = html_manga.body().requireElementById("about").text()
+
+		return manga.copy(
+			description = about,
+			chapters = listOf(
+				MangaChapter(
+					id = generateUid(manga.id),
+					name = manga.title,
+					number = 1,
+					url = manga.url,
+					scanlator = null,
+					uploadDate = date.tryParse(json!!.getString("add_date")),
+					branch = null,
+					source = source,
+				)
+			)
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		// No pagination
+		if(page != 1){
+			return emptyList()
+		}
+
+		// Get all manga
+		var allManga = webClient.httpGet(allManga).parseJson().getJSONArray("manga").toJSONList()
```

Comment:
Looks like result of this call might be cached for faster `getDetails` evaluation. Additionally it is better to use `SoftReference` for it since it may be very big